assets/src/components/with-cropped-featured-image.js

```diff
@@ -0,0 +1,264 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { dispatch } from '@wordpress/data';
+
+/**
+ * External dependencies
+ */
+import { now } from 'lodash';
+
+const EXPECTED_WIDTH = 696;
+const EXPECTED_HEIGHT = 928;
+
+/**
+ * Gets a wrapped version of MediaUpload.
+ *
+ * Only applies to the MediaUpload in the Featured Image component, PostFeaturedImage.
+ * Suggests cropping of the featured image if it's not 696 x 928.
+ * Mostly copied from customize-controls.js.
+ *
+ * @param {Function} InitialMediaUpload The MediaUpload component, passed from the filter.
+ * @return {Function} The wrapped component.
+ */
+export default ( InitialMediaUpload ) => {
+	/**
+	 * Mostly copied from customize-controls.js, with slight changes.
```

Comment:
This mainly copies functions from [wp.customize.CroppedImageControl](https://github.com/WordPress/wordpress-develop/blob/master/src/js/_enqueues/wp/customize/controls.js#L4495).

It would probably be better to use functions from that directly, instead of mainly copying them. Like with `wp.customize.CroppedImageControl.prototype.mustBeCropped()`.

But `wp.customize` doesn't look to be available in the block editor.

Reply:
Unfortunately I am not really an expert on `wp.media` 😕 

I think in this case it is fine to copy over the functions. Enqueueing customizer scripts in the editor would probably result in too much bloat and maybe even some unwanted side effects.

Reply:
Yeah, good point.